### ISSUE TITLE
need help to send last image through MQTT

### ISSUE THREAD

**Issue by Bloodboy1986** [jomjol/AI-on-the-edge-device#1102]
Hi guys,

this project is amazing :)

I'm currently trying to send the current image via mqtt and already have a [proof of concept](https://github.com/Bloodboy1986/AI-on-the-edge-device/commit/b5017cea13328005e615074eccdbd7071f46e6bc
) for it.

However, I still have the problem that I can not find the current image. At the moment I am still using a static test image.
I know that the current image can be retrieved via the URL /img_tmp/alg_roi.jpg. However, I can't find it on the sdcard.
Maybe one of you would be so kind and help me with this? :)

My c/c++ knowledge is not the best either...

If you interesset i will create a PR later on.

**Comment by caco3** [comment]
I think the image does not get stored to the sd-card at all.

As far as I can see, the image gets provided on the fly, see https://github.com/jomjol/AI-on-the-edge-device/blob/rolling/code/components/jomjol_flowcontroll/ClassFlowControll.cpp#L579
You might be able to call function `GetJPGStream()` like in https://github.com/jomjol/AI-on-the-edge-device/blob/rolling/code/components/jomjol_tfliteclass/server_tflite.cpp#L51.

**Comment by jomjol** [comment]
@caco3 is totally right. Background is, that we want to avoid too much sd-card reading, as this has created a lot of trouble in the past.
If you want to change this, you need to modify the code. You you can also activly pull the image via `http://IP-ADRESS/img_tmp/raw.jpg`  or  `.../alg.jpg`   or `.../alg_roi.jpg`

**Comment by Bloodboy1986** [comment]
Thanks for your answers :)

I already tried retrieving the image directly , however it seems to load the esp a lot when doing so, especially when it's processing. It has often happened that I could not retrieve the image.
So I thought it would be easy to send the current image via mqtt after the flow has finished.

So you thing, i can just call GetJPGStream() to get the actual image?
I think i dont need the image as file....

I would be very happy to have your help :)

**Comment by jomjol** [comment]
No, the function `esp_err_t ClassFlowControll::GetJPGStream(std::string _fn, httpd_req_t *req)` is directly sending an image to http-request (req).
There is no function, where you can directly get the image as an result.

**Comment by caco3** [comment]
Is this function used? https://github.com/jomjol/AI-on-the-edge-device/blob/aaad9527827128b92a4fbcb3f66fddeb02b54b35/code/components/jomjol_flowcontroll/ClassFlowAlignment.cpp#L159

There is a function https://github.com/jomjol/AI-on-the-edge-device/blob/aaad9527827128b92a4fbcb3f66fddeb02b54b35/code/components/jomjol_flowcontroll/ClassFlowAlignment.cpp#L207

I would **not** save it to the sd-card as this would destroy the sd-card quickly!
How ever you maybe could store the image in a global variable and use it from there.

**Comment by Bloodboy1986** [comment]
Hello guys,

sorry for my late feedback, but in Germany was a long weekend due to a holiday and I was on vacation :)

I have now looked at the whole thing again and as a workaround and plan B I let first write the current image on the sdcard. But I also see it as very problematic. The only way out would be to use an ENDURANCE or INDUSTRIAL sdcard...

I must admit that I do not get warm with c++ :(
I think I understood that CImageBasis is some kind of construct which contains the current image. I created my own function which returns me only the CImageBasis, just like in "GetJPGStream", only without the http part. Only how I process this now, I'm really overwhelmed....

Maybe one of you has once again time and interest to help me :)
At the end I need the current image as filestream in binary form.
